driver: usb_dc_sam_usbhs: Restore EP0 after detach

desowin opened the pull request:
SAM V71 Xplained Ultra failed to re-enumerate on the bus after device detach-attach sequence. The device was recognized by the host, but it was not responding to control transfers, i.e. it was not acknowledging SETUP data. Make device acknowledge SETUP data by reconfiguring EP0 when necessary and actually set DETACH bit during detach instead of clearing it.

Fixes  #50549

tmon-nordic committed:
driver: usb_dc_sam_usbhs: Restore EP0 after detach

SAM V71 Xplained Ultra failed to re-enumerate on the bus after device
detach-attach sequence. The device was recognized by the host, but it
was not responding to control transfers, i.e. it was not acknowledging
SETUP data. Make device acknowledge SETUP data by reconfiguring EP0
when necessary.

Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>

anangl commented on drivers/usb/device/usb_dc_sam_usbhs.c:
```diff
@@ -348,7 +358,7 @@ int usb_dc_attach(void)
 {
 	/* Detach the device */
-	USBHS->USBHS_DEVCTRL &= ~USBHS_DEVCTRL_DETACH;
+	USBHS->USBHS_DEVCTRL |= USBHS_DEVCTRL_DETACH;
```
This change seems to deserve a separate commit.

desowin replied:
Done.

tmon-nordic committed:
driver: usb_dc_sam_usbhs: Actually set detach bit

Device physically detaches when the DETACH bit is set. Fix the
assignment to actually set the bit instead of clearing it.

Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>
